### DISCUSSION TITLE
Obtaining the URL that led to a 404 not-found page, from the layout

### DISCUSSION THREAD

**Discussion by Ferk** [vuejs/vitepress#2452] [Q&A]
I'm using the `#not-found` [layout slot ](https://vitepress.dev/guide/extending-default-theme#layout-slots) from the default theme to change the 404 page.

However, I'd like to write some logic that would use the name of the missing page to change what's displayed in the not-found template (for example, a link to Gitlab for making a PR and add the missing page).

I'm not sure if there's a way, at present, to access this information in the way it's currently set up in Vitepress.

It seems the `page` property from `useData()` doesn't contain much information when the page is not found.

```{ "relativePath": "", "filePath": "", "title": "404", "description": "Not Found", "headers": [], "frontmatter": { "sidebar": false, "layout": "page" }, "lastUpdated": 0, "isNotFound": true }```

Is this a missing feature, or if there an alternative way to do this?

Trying to read the URL from `window.location.href` is a bit hacky, and it doesn't seem to work after building the site statically.

**Comment by alephpi** [comment]
maybe `useRoute().path` is what you need?
